### ISSUE TITLE
viztracer stop monitor script before the script ends

### ISSUE THREAD

**Issue by taicaile** [gaogaotiantian/viztracer#121]
Hello,

I am trying to monitor the an async script with command
```bash
viztracer script.py --log_async
```
The log is shown below:

```text
[I:pyppeteer.launcher] Browser listening on: ws://127.0.0.1:48413/devtools/browser/f55f03f4-d402-468e-a3f5-7c4965c0a12d
title:Example Domain
[I:pyppeteer.launcher] terminate chrome process...
17:27:59,141 INFO: Received exit signal shutdown...
17:27:59,142 INFO: Cancelling 0 outstanding tasks
17:27:59,142 INFO: Gathering 0 tasks
17:27:59,142 INFO: Flushing metrics
Loading finish                                        
==================================================
== Starting from version 0.13.0, VizTracer will ==
== use json as the default report file. You can ==
== generate HTML report with "-o result.html"   ==
==================================================
Dumping trace data, total entries: 517717
Saving report to /mnt/sda1/github-own/python-study/coroutine/result.json ...
Use "vizviewer <your_report>" to open the report
[I:pyppeteer.launcher] terminate chrome process...
```
The last line was printed by the script, which should have been printed before **viztracer** exited.

**Comment by gaogaotiantian** [comment]
Could you put up a minimum example to reproduce this? I noticed that there's another line of ```[I:pyppeteer.launcher] terminate chrome process...``` before VizTracer dumped the data. Did you have two chrome processes? Have you checked the report at the end and confirmed the termination was ever executed?

Pyppeteer has multiple exit routines registered just like VizTracer so there might be some conflicts. It's hard to tell what went wrong(or if anything went wrong) without some script to reproduce.

**Comment by taicaile** [comment]
Hello,

```python
import asyncio
import functools
import os
import signal
from typing import overload

import asyncio
from pyppeteer import launch

import logging

logging.basicConfig(
    level=logging.INFO,
    format="%(asctime)s,%(msecs)d %(levelname)s: %(message)s",
    datefmt="%H:%M:%S",
)

async def shutdown(signal, loop):
    """Cleanup tasks tied to the service's shutdown."""
    logging.info(f"Received exit signal {signal}...")
    tasks = [t for t in asyncio.all_tasks() if t is not
             asyncio.current_task()]
    [task.cancel() for task in tasks]
    logging.info(f"Cancelling {len(tasks)} outstanding tasks")
    await asyncio.gather(*tasks)
    logging.info(f'Gathering {len(tasks)} tasks')
    logging.info(f"Flushing metrics")

async def main():
    browser = await launch(headless=False,
                           userDataDir='UserDataDir',
                           handleSIGTERM=False,
                           handleSIGINT=False,)
    page = await browser.newPage()
    await page.goto('http://example.com')
    title = await page.title()
    print(f"title:{title}")
    await browser.close()

@overload
def signal_handler(signum, frame): ...
def signal_handler(name, frame=None):
    print('signal_handler({!r})'.format(name))

event_loop = asyncio.get_event_loop()

for name in ['SIGHUP', 'SIGINT']:
    event_loop.add_signal_handler(
        getattr(signal,name),
        functools.partial(signal_handler, name=name),
    )
    signal.signal(getattr(signal,name),signal_handler)

async def send_signals():
    pid = os.getpid()
    print('starting send_signals for {}'.format(pid))

    for name in ['SIGHUP', 'SIGHUP', 'SIGINT']:
        print('---- sending {} ----'.format(name))
        os.kill(pid, getattr(signal, name))
        print('yielding control')
        await asyncio.sleep(0.01)
    return

try:
    event_loop.run_until_complete(main())
    # event_loop.run_until_complete(send_signals())
finally:
    asyncio.run(shutdown('shutdown', event_loop))
```

**Comment by gaogaotiantian** [comment]
So the reason this happens is because pyppeteer registered an ``atexit`` function, but viztracer stopped tracing and dumped data before exit the program. The ``atexit`` function was called right before VizTracer exited(but after dumping the report).

In this specific case, it has little impact because your chrome process was shutdown already by ``browser.close()`` function. However, there may be cases where people want to track down the ``atexit`` functions with VizTracer.

I already put up a fix in #123 and it should be merged in soon.
